src/controllers/AuthController.js

```diff
@@ -29,28 +26,50 @@ const userToken = (user) => {
 };
 
 const AuthController = {
-  async signin(req, res, next) {
+  async signin(req, res) {
     try {
-      // extracting user data
       const { email, password, username } = req.body;
-
-      // checking if the user exist
       const user = email ? await User.findOne({ where: { email } })
         : await User.findOne({ where: { username } });
 
       if (!user) return sendErrorResponse(res, 404, 'User Not Found!!');
 
-      // compare password
       const checkPassword = comparePassword(password, user.dataValues.password);
       if (!checkPassword) return sendErrorResponse(res, 400, 'Email or Password does not match');
 
-      // check user verification
       if (!user.dataValues.verified) return sendErrorResponse(res, 401, 'Verify Your Account ');
       const createdToken = userToken(user.dataValues);
 
       return sendSuccessResponse(res, 200, createdToken);
     } catch (e) {
-      return next(e);
+      return sendErrorResponse(res, 500, 'INTERNAL SERVER ERROR');
+    }
+  },
+  async signup(req, res) {
+    try {
+      const {
+        firstName, surName, lastName, email, phone, conditions, role
+      } = req.body;
+      let { password } = req.body;
+      password = hashPassword(password);
+
+      if (!req.file) {
+        return sendSuccessResponse(res, 422, 'Please select an image file to upload');
```

Comment:
For now, no image will be required on sign up, for better user experience. profiles will be completed on user dashboards.

Reply:
It was part of the fields in the spec you assigned me. It's why you're seeing it. But you're right, it's not necessary on sign-up.